client.go

```diff
@@ -108,6 +112,189 @@ func (c *Client) SharedConfig(ctx context.Context) (*response.SharedConfig, erro
+// UpdateTransactionMetadataRequest for SPV Wallet API consumption.
+func (u UpdateTransactionMetadataArgs) ParseUpdateTransactionMetadataRequest() transactions.UpdateTransactionMetadataRequest {
+	return transactions.UpdateTransactionMetadataRequest{
+		ID:       u.ID,
+		Metadata: u.Metadata,
+	}
+}
+
+// DraftTransaction creates a new draft transaction using the user transactions API.
+// This method sends an HTTP POST request to the "/draft" endpoint and expects
+// a response that can be unmarshaled into a response.DraftTransaction struct.
+// If the request fails or the response cannot be decoded, an error is returned.
+func (c *Client) DraftTransaction(ctx context.Context, args DraftTransactionArgs) (*response.DraftTransaction, error) {
+	res, err := c.transactionsAPI.DraftTransaction(ctx, args.ParseToDraftTransactionRequest())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create draft transaction by call user transactions API: %w", err)
+	}
+
+	return res, nil
+}
+
+// RecordTransaction submits a transaction for recording using the user transactions API.
+// This method sends an HTTP POST request to the "/transactions" endpoint, expecting
+// a response that can be unmarshaled into a response.Transaction struct.
+// If the request fails or the response cannot be decoded, an error is returned.
+func (c *Client) RecordTransaction(ctx context.Context, args RecordTransactionArgs) (*response.Transaction, error) {
+	res, err := c.transactionsAPI.RecordTransaction(ctx, args.ParseToRecordTransactionRequest())
+	if err != nil {
+		return nil, fmt.Errorf("failed to record transaction with reference ID: %s by call user transactions API: %w", args.ReferenceID, err)
+	}
+
+	return res, nil
+}
+
+// UpdateTransactionMetadata updates the metadata of a transaction using the user transactions API.
+// This method sends an HTTP PATCH request with updated metadata and expects a response
+// that can be unmarshaled into a response.Transaction struct.
+// If the request fails or the response cannot be decoded, an error is returned.
+func (c *Client) UpdateTransactionMetadata(ctx context.Context, args UpdateTransactionMetadataArgs) (*response.Transaction, error) {
+	res, err := c.transactionsAPI.UpdateTransactionMetadata(ctx, args.ParseUpdateTransactionMetadataRequest())
+	if err != nil {
+		return nil, fmt.Errorf("failed to update transactions metadata by call user user transactions API: %w", err)
+	}
+
+	return res, nil
+}
+
+// Transactions retrieves a list of transactions using the user transactions API.
+// This method applies optional query parameters and expects a response that can be
+// unmarshaled into a slice of response.Transaction pointers.
+// If the request fails or the response cannot be decoded, an error is returned.
+func (c *Client) Transactions(ctx context.Context, opts ...QueryOption) ([]*response.Transaction, error) {
+	var params Query
+	params.apply(opts...)
```

Comment:
```suggestion
	var params queryParams
	params.apply(opts...)
```
If the `apply` (and other its methods, I suppose) are unexported - maybe the `Query` itself should be unexported? 🤔

Reply:
I have prepared a completely different approach :) I will explain everything during tomorrow's conversation.